File: frontend/src/features/auth/model/UserProfileResponse.ts
import type { DisplayName, Username } from '@/features/auth/model/AuthValues'
import type { UserAcceptedProblem } from '@/features/auth/model/UserAcceptedProblem'
import type { UserContribution } from '@/features/auth/model/UserContribution'

export type UserProfileResponse = {
  username: Username
  displayName: DisplayName
  contribution: UserContribution
  acceptedProblems: UserAcceptedProblem[]
}

File: backend/src/main/scala/domains/auth/model/UserProfileResponse.scala
package domains.auth.model

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UserProfileResponse(
  username: Username,
  displayName: DisplayName,
  contribution: UserContribution,
  acceptedProblems: List[UserAcceptedProblem]
)

object UserProfileResponse:
  given Encoder[UserProfileResponse] = deriveEncoder[UserProfileResponse]
  given Decoder[UserProfileResponse] = deriveDecoder[UserProfileResponse]
